lib/index.coffee

```diff
@@ -72,7 +73,7 @@ class DockerGC
 	setDocker: (hostObj) ->
 		@currentMtimes = {}
 		@hostObj = _.defaults({ Promise }, hostObj)
-		@dockerProgress = new DockerProgress(@hostObj)
+		@dockerProgress = new DockerProgress({ docker: new Docker(@hostObj) })
```

Comment:
* This changes adapts to the dependency injection(\*) requirement of `docker-progress` v5, providing an instance of `dockerode` instead of having `docker-progress` create its own instance (which was the behaviour before this PR). This avoids `docker-progress` using a different version of `dockerode` (e.g. v2 vs v3) than its caller (`docker-storage-gc`, `resin-multibuild`, `balena-cli`, `balena-builder`, `balena-preload`, etc.), with multiple copies of `dockerode` under the `node_modules` folder.

(\*) Re dependency injection, there is the theoretical risk of injecting an instance of dockerode whose public interface is so different that it breaks things. However:
* Where TypeScript is used, this should be caught at compile time. `balena-buider` and `balena-cli` use TypeScript, even though `docker-storage-gc` doesn't.
* `docker-progress` makes limited use of `dockerode`, basically `pull()`, `push()` and `version()`, so as long as these methods have sufficiently similar interfaces, it is fine.
* I have tested with `dockerode` v2 and v3, didn't come across issues.

My reading of the code tells me that we could have reused the `dockerode` instance fetched with `getDocker` just one line below, instead of calling `new Docker(@hostObj)`. But this could subtly change existing behaviour and I really don't want to rock the boat on this one! :-)  (`docker-storage-gc` was the subject of intricate bugs and investigations in the past.)

By the way, do we even need to use `docker-progress` in `docker-storage-gc`? The only call is on this line:
https://github.com/balena-io-modules/docker-storage-gc/blob/v3.5.0/lib/index.coffee#L147
```
@dockerProgress.pull(baseImage, _.noop)
```

I understand that `docker-progress` provides for nicer download percentage reporting to the console, but is it even displayed anywhere in the case of `docker-storage-gc`? If not, we could use `dockerode`'s `pull()` method directly, instead of using `docker-progress`. Fewer dependencies, less code, less likely to go wrong...